src/lib.rs

```diff
@@ -168,7 +171,87 @@ mod arch {
     }
 }
 
-#[cfg(not(target_os = "linux"))]
+#[cfg(target_os = "macos")]
+mod arch {
+    use super::*;
+
+    pub const SPAWN_ERROR: &str = "could not spawn xctrace";
+    pub const WAIT_ERROR: &str = "unable to wait for xctrace record child command to exit";
+
+    pub(crate) fn initial_command(
+        workload: Workload,
+        sudo: Option<Option<&str>>,
+        _freq: u32,
+        _custom_cmd: Option<String>,
+        verbose: bool,
+        ignore_status: bool,
+    ) -> Option<PathBuf> {
+        let xctrace = env::var("XCTRACE").unwrap_or_else(|_| "xctrace".to_string());
+        let trace_file = PathBuf::from("cargo-flamegraph.trace");
+        let mut command = sudo_command(&xctrace, sudo);
+        command
+            .arg("record")
+            .arg("--template")
+            .arg("Time Profiler")
+            .arg("--target-stdout")
+            .arg("-")
+            .arg("--output")
+            .arg(&trace_file);
+        match workload {
+            Workload::Command(args) => {
+                command.arg("--launch").arg("--").args(args);
+            }
+            Workload::Pid(pid) => {
+                match &*pid {
+                    [pid] => {
+                        // xctrace could accept multiple --attach <pid> arguments,
+                        // but it will only profiling on the last pid provided.
+                        command.arg("--attach").arg(pid.to_string());
+                    }
+                    _ => {
+                        panic!("xctrace only supports profiling a single process at a time");
+                    }
+                }
+            }
+            Workload::ReadPerf(_) => {}
+        }
+        run(command, verbose, ignore_status);
+        Some(trace_file)
+    }
+
+    pub fn output(
+        trace_file: Option<PathBuf>,
+        script_no_inline: bool,
+        _sudo: Option<Option<&str>>,
+    ) -> anyhow::Result<Vec<u8>> {
+        if script_no_inline {
+            return Err(anyhow::anyhow!("--no-inline is only supported on Linux"));
+        }
+
+        let xctrace = env::var("XCTRACE").unwrap_or_else(|_| "xctrace".to_string());
+        let trace_file = trace_file.context("No trace file found.")?;
+        let output = Command::new(xctrace)
+            .arg("export")
+            .arg("--input")
+            .arg(&trace_file)
+            .arg("--xpath")
+            .arg(r#"/trace-toc/*/data/table[@schema="time-profile"]"#)
+            .output()
+            .context("Run xctrace export failed.")?;
+        std::fs::remove_dir_all(&trace_file)
+            .with_context(|| anyhow!("Remove trace({}) failed.", trace_file.to_string_lossy()))?;
```

Comment:
Nit: please don't capitalize the error, for consistency -- also, skip the period at the end. (Same for the "Run xctrace export failed" above.)